### PR TITLE
feat: Add support for Managed Node Groups (`node_groups`) taints

### LINKED ISSUE #1403: aws_eks_node_group supports taints from aws provider version 3.43.0
## Is your request related to a new offering from AWS?

yes, scheduled for release 3.43.0 
https://github.com/hashicorp/terraform-provider-aws/issues/19310

## Is your request related to a problem? Please describe.

Will solve workaround with `kubelet_extra_args` to add taints to nodes
and relates to some issues:
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1232
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1214
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/1085
https://github.com/terraform-aws-modules/terraform-aws-eks/issues/962

## Describe the solution you'd like.

Simply set the block and make it available as input variable similar to `k8s_labels`

```
resource "aws_eks_node_group" "example" {
  ...

  taint {
    key    = "dedicated"
    value  = "gpuGroup"
    effect = "NO_SCHEDULE"
  }

  taint {
    key    = "special"
    value  = "true"
    effect = "PREFER_NO_SCHEDULE"
  }
}
```

### PR DESCRIPTION
**Pull request by nauxliu**
## Description

Add support for set taints on managed node groups, Resolve #1403

Ref:
https://docs.aws.amazon.com/eks/latest/userguide/node-taints-managed-node-groups.html
https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#taint-configuration-block

### Checklist

- [x] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

File: modules/node_groups/node_groups.tf [hunk 1]
```diff
@@ -57,6 +57,16 @@ resource "aws_eks_node_group" "workers" {
     }
   }
 
+  dynamic "taint" {
+    for_each = each.value["taints"] == null ? [] : each.value["taints"]
```
Comment (barryib) [inline modules/node_groups/node_groups.tf:61]: Can we just define `taints = []` default value in https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/modules/node_groups/locals.tf#L5-L23

Comment (nauxliu) [reply modules/node_groups/node_groups.tf:61]: Why? I think no one will to define taints in workers_group_defaults to apply taints to all node_groups.

Comment (nauxliu) [reply modules/node_groups/node_groups.tf:61]: Sorry, I misunderstood the role of the node_groups_expanded,  Thanks for the suggestion, I'll change it.

Comment (barryib) [reply modules/node_groups/node_groups.tf:61]: > Why? I think no one will define taints in workers_group_defaults to apply taints to all node_groups

You're absolutely right. It's just for consistency with other arguments. That local, is used most of the time to know what are available arguments. And it's also for simplicity.

Comment (nauxliu) [reply modules/node_groups/node_groups.tf]: force pushed

**Review by barryib — APPROVED** [review]

**Comment by barryib** [comment]
Thanks again @nauxliu for your contribution.

**Comment by karlmuscat** [comment]
@barryib any idea when is this merge getting released? thank you :)

**Comment by mykolasmith** [comment]
+1 ^

**Comment by barryib** [comment]
Just released it in [v17.1.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/tag/v17.1.0)

**Pull request merged**